src/components/TextInput/TextInput.stories.mdx

```diff
@@ -204,6 +205,48 @@ within the `label` element.
   </DSProvider>
 </Canvas>
 
+## Number Type
+
+The `TextInput` component can be configured to render a number input by setting
+`type="number"`. In this input type, the `maxLength` prop does not restrict the
+length of the number that is passed to the input. This is the expected behavior
+as the native HTML input and `maxlength` attribute behave this way.
+
+For this type, the `min` and `max` props do restrict the value of the number
+input that is passed. This can be seen by using the up and down arrows to
+increase or decrease the number value, respectively. However, if the `max` prop
+is set to "200", for example, the user can still enter a number such as "400".
+The consuming application must handle validating the entered number value.
+
+If the `min` prop value is greater than the `max` prop value, it will trigger
+the invalid state.
+
+<Canvas>
+  <DSProvider>
+    <VStack align="stretch" spacing={12}>
+      <TextInput
+        helperText="The min value is 10 and the max value is 20."
+        id="number-type"
+        labelText="Enter a number"
+        max={20}
+        min={10}
+        placeholder="15"
```

Comment:
When I saw the `15` in the input, I was confused and wondered why it was italicized.  It did not occur to me that I was looking at the placeholder.  Let’s follow the same pattern as the other inputs on the screen and use `i.e. 1, 2, 3, etc.`.